src/components/DayPicker.jsx

```diff
@@ -478,16 +521,31 @@ class DayPicker extends React.Component {
     return focusedDate;
   }
 
-  setCalendarMonthHeights(calendarMonthHeights) {
+  setMonthTitleHeight(monthTitleHeight) {
+    const { daySize } = this.props;
+    const calendarMonthWeeksHeight = Math.max(0, ...this.calendarMonthWeeks) * (daySize - 1);
+    const newMonthHeight = monthTitleHeight + calendarMonthWeeksHeight + 1;
+
+    if (this.isHorizontal()) {
+      this.adjustDayPickerHeight(newMonthHeight);
+    }
+
+    this.setState({
+      monthTitleHeight,
+      hasSetHeight: true,
+    });
+  }
+
+  setCalendarMonthWeeks() {
     const { numberOfMonths } = this.props;
-    const firstVisibleMonthIndex = this.getFirstVisibleIndex();
-    const lastVisibleMonthIndex = firstVisibleMonthIndex + numberOfMonths;
+    const { currentMonth } = this.state;
 
-    this.calendarMonthHeights = calendarMonthHeights;
-    const visibleCalendarMonthHeights = calendarMonthHeights
-      .filter((_, i) => ((i >= firstVisibleMonthIndex) && (i < lastVisibleMonthIndex)));
-    this.calendarMonthGridHeight = Math.max(0, ...visibleCalendarMonthHeights) + MONTH_PADDING;
-    this.setState({ hasSetHeight: true });
+    this.calendarMonthWeeks = [];
+    for (let i = 0; i < numberOfMonths; i += 1) {
```

Comment:
I wonder if we could tweak this to in the same way to capture months before and after the current month? And then on prev/next click, we could update the neighbor months asynchronously, the way that we might preload images in a carousel? I could be misunderstanding the logic too.